sources/packages/api/src/route-controllers/restriction/restriction.controller.ts

```diff
@@ -233,4 +238,185 @@ export class RestrictionController extends BaseController {
+    };
+  }
+
+  /**
+   * Rest API to add a new provincial restriction to Institution.
+   * @param userToken
+   * @param institutionId
+   * @param restrictionId
+   * @param payload
+   */
+  @Groups(UserGroups.AESTUser)
+  @AllowAuthorizedParty(AuthorizedParties.aest)
+  @Post("/institution/:institutionId")
+  async addInstitutionProvincialRestriction(
+    @UserToken() userToken: IUserToken,
+    @Param("institutionId") institutionId: number,
+    @Body() payload: AssignRestrictionDTO,
+  ): Promise<void> {
+    const restriction =
+      await this.restrictionService.getProvincialRestrictionById(
+        payload.restrictionId,
+        true,
+      );
+    if (!restriction) {
+      throw new UnprocessableEntityException(
+        "The given restriction type is either not Provincial or  not Institution.",
+      );
+    }
+    const updatedRestriction =
+      await this.institutionRestrictionService.addProvincialRestriction(
+        institutionId,
+        userToken.userId,
+        payload,
+      );
+    /**mapping the note added for restriction to institution notes**/
+    if (updatedRestriction.restrictionNote) {
+      await this.institutionService.saveInstitutionNote(
+        institutionId,
+        updatedRestriction.restrictionNote,
+      );
+    }
+  }
+
+  /**
+   * Rest API to resolve a provincial restriction for Institution.
+   * @param userToken
+   * @param institutionId
+   * @param institutionRestrictionId
+   * @param payload
+   */
+  @Groups(UserGroups.AESTUser)
+  @AllowAuthorizedParty(AuthorizedParties.aest)
+  @Patch(
+    "/institution/:institutionId/institutionRestriction/:institutionRestrictionId/resolve",
+  )
+  async resolveInstitutionProvincialRestriction(
+    @UserToken() userToken: IUserToken,
+    @Param("institutionId") institutionId: number,
+    @Param("institutionRestrictionId") institutionRestrictionId: number,
+    @Body() payload: ResolveRestrictionDTO,
+  ): Promise<void> {
+    if (!payload.noteDescription) {
+      throw new UnprocessableEntityException(
+        "Resolution Notes are mandatory to resolve the restriction.",
+      );
+    }
+    try {
+      const updatedRestriction =
+        await this.institutionRestrictionService.resolveProvincialRestriction(
+          institutionId,
+          institutionRestrictionId,
+          userToken.userId,
+          payload,
+        );
+
+      /**mapping the note added for resolution to student notes**/
+      await this.institutionService.saveInstitutionNote(
+        institutionId,
+        updatedRestriction.resolutionNote,
+      );
+    } catch (error) {
+      if (error.name === RESTRICTION_NOT_ACTIVE) {
+        throw new UnprocessableEntityException(error.message);
+      }
+      throw new InternalServerErrorException(
+        "Unexpected error while resolving restriction",
+      );
+    }
+  }
+
+  /**
+   * Rest API to get restriction status for an institution.
+   * @param institutionId
```

Comment:
is this and 
![image](https://user-images.githubusercontent.com/77353155/148860803-309858d6-88a2-4ce5-b9a8-ec16c02ff9a5.png)
 are different, i didn't get this part

Reply:
Line no 408 is the API which tells if the institution has any valid restrictions. This API is for future purpose show the badge in UI(Active restriction badge) or to block the actions of institution if it had restrictions to it.

One in the screenshot is a plain get by id API.